### DISCUSSION TITLE
Tutorial/Example of two separate processes communicating using RustDDS?

### DISCUSSION THREAD

**Discussion by IndianBoy42** [Atostek/RustDDS#168] [Q&A]
Does anyone have an annotated example/tutorial about how to write 2 rust programs that discover and communicate with each other using RustDDS?

Within the docs.rs of rustdds there is an example that just communicates within the same process, and just separating those into 2 different programs hasn't worked for me. Should it?

There is [this](https://github.com/jhelovuo/RustDDS/blob/master/examples/shapes_demo/main.rs) example which communicates with other processes, and has to setup a lot of mio stuff. Is that required? This example also seems to be out of date

**Comment by jhelovuo** [comment]
This answer is probably a bit late, but here goes anyway:

Yes, communication across two different processes should work just fine.

You sort of have to use the mio stuff to know when data has arrived to a DataReader. 

If you do not, then you have to set up a polling loop to periodically check for data arrival. The read/take calls are asynchronous so that they always return immediately, returning any data that was available, or reporting that there is nothing. Currently, there is no synchronous interface that would block your thread to wait for incoming data.
